File: 020/Haskell/Problem020Test.hs
module Main where

import Test.HUnit
import System
import Problem020

sumOfDigitsOfFactorialTests = TestList [
  TestCase (1 @=? sumOfDigitsOfFactorial 1),
  TestCase (6 @=? sumOfDigitsOfFactorial 3),
  TestCase (3 @=? sumOfDigitsOfFactorial 5),
  TestCase (9 @=? sumOfDigitsOfFactorial 6),
  TestCase (1 @=? 1)
  ]
  
sumOfDigitsTests = TestList [
  TestCase (1 @=? sumOfDigits 1),
  TestCase (3 @=? sumOfDigits 12),
  TestCase (14 @=? sumOfDigits 86),
  TestCase (19 @=? sumOfDigits 1945),
  TestCase (1 @=? 1)
  ]
  
tests = TestList [
	TestLabel "testing sumOfDigitsOfFactorial" sumOfDigitsOfFactorialTests,
	TestLabel "testing sumOfDigits" sumOfDigitsTests,
	TestCase (1 @=? 1)
	]
	
main = runTestTT tests >>= checkTests
  where
    checkTests counts = case counts of
    	Counts _ _ errors failures ->
    		if (errors + failures > 0)
    		then exitFailure
    		else (exitWith ExitSuccess)

File: 020/Ruby/problem020_test.rb
require "test/unit"

require_relative "problem020"

class TestProblem020 < Test::Unit::TestCase

  def test_sum_of_digits_of_factorial
    assert_equal 1, 1.sum_of_digits_of_factorial
    assert_equal 6, 3.sum_of_digits_of_factorial
    assert_equal 3, 5.sum_of_digits_of_factorial
    assert_equal 9, 6.sum_of_digits_of_factorial
  end

  def test_sum_of_digits
    assert_equal 1, 1.sum_of_digits
    assert_equal 3, 12.sum_of_digits
    assert_equal 14, 86.sum_of_digits
    assert_equal 19, 1945.sum_of_digits
  end

end
